src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -263,6 +263,29 @@ public ValueTask ConnectAsync(string host, int port, CancellationToken cancellat
             return ConnectAsync(ep, cancellationToken);
         }
 
+        /// <summary>
+        /// Disconnects a connected socket from the remote host.
+        /// </summary>
+        /// <param name="reuseSocket">Indicates whether the socket should be available for reuse after disconnect.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used to cancel the asynchronous operation.</param>
+        /// <returns>An asynchronous task that completes when the socket is disconnected.</returns>
+        public ValueTask DisconnectAsync(bool reuseSocket, CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return ValueTask.FromCanceled(cancellationToken);
+            }
+
+            AwaitableSocketAsyncEventArgs saea =
+                Interlocked.Exchange(ref _singleBufferSendEventArgs, null) ??
+                new AwaitableSocketAsyncEventArgs(this, isReceiveForCaching: false);
+
+            saea.DisconnectReuseSocket = reuseSocket;
+            saea.WrapExceptionsForNetworkStream = false;
```

Comment:
I was about to comment that this line shouldn't be necessary, but I see from the implementation that it is, which makes me think we should also be setting this here:
https://github.com/dotnet/runtime/blob/d776e5b6d7e46eac4713ed31ff17a22ec0329f25/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs#L111
It's super rare that it would be an issue, as it would require disconnecting and then reconnecting the same socket, end up using the same args instance that was previously used via NetworkStream.  But, still, probably worth fixing.